qlty-config/src/config/builder.rs

```diff
@@ -355,9 +358,97 @@ impl Builder {
     }
 }
 
+fn merge_enabled_plugins(existing: &EnabledPlugin, new: &EnabledPlugin) -> EnabledPlugin {
+    debug!(
+        "Merging enabled plugins: existing: {:?}, new: {:?}",
+        existing, new
+    );
+
+    if existing.version != new.version {
+        warn!(
+            "Merging enabled plugins with different versions: {} and {}, using {}.",
+            existing.version, new.version, new.version
+        );
+    }
+
+    let merged_mode = existing.mode.unwrap_or(new.mode.unwrap_or_default());
+
+    EnabledPlugin {
+        name: existing.name.clone(),
+        prefix: existing.prefix.clone(),
+        mode: Some(merged_mode),
+        version: new.version.clone(),
+        triggers: existing
+            .triggers
+            .iter()
+            .chain(new.triggers.iter())
+            .cloned()
+            .collect(),
+        skip_upstream: existing.skip_upstream.or(new.skip_upstream),
+        package_file: existing.package_file.clone().or(new.package_file.clone()),
+        fetch: existing
+            .fetch
+            .iter()
+            .chain(new.fetch.iter())
+            .cloned()
+            .collect(),
+        package_filters: existing
+            .package_filters
+            .iter()
+            .chain(new.package_filters.iter())
+            .cloned()
+            .collect(),
+        affects_cache: existing
+            .affects_cache
+            .iter()
+            .chain(new.affects_cache.iter())
+            .cloned()
+            .collect(),
+        extra_packages: existing
+            .extra_packages
+            .iter()
+            .chain(new.extra_packages.iter())
+            .cloned()
+            .collect(),
+        drivers: existing
+            .drivers
+            .iter()
+            .chain(new.drivers.iter())
+            .cloned()
+            .collect(),
+        config_files: existing
+            .config_files
+            .iter()
+            .chain(new.config_files.iter())
+            .cloned()
+            .collect(),
+    }
+}
+
+fn compute_unique_merged_enabled_plugins(plugins: &[EnabledPlugin]) -> Vec<EnabledPlugin> {
+    let mut merged_unique_enabled_plugins: HashMap<(String, String), EnabledPlugin> =
+        HashMap::new();
+
+    for plugin in plugins.iter() {
+        let key = (
+            plugin.name.clone(),
+            plugin.prefix.clone().unwrap_or_default(),
```

Comment:
Prefix is selected here for generating the keying for unique for plugin name and prefix.